### PR TITLE
YJIT: Leave cfp->pc uninitialized for VM_FRAME_MAGIC_CFUNC

### PR DESCRIPTION
**Pull request by XrXr**
C function frames don't need to use the VM-specific pc field to run
properly. When pushing a control frame from output code, save one
instruction by leaving the field uninitialized.

Fix-up rb_vm_svar_lep(), which is used while setting local variables via
Regexp#=~. Use cfp->iseq as a secondary signal so it can stop assuming
that all CFUNC frames always have zero pc's.

### CODE REVIEW

**Comment by XrXr** [comment]
Note, this shouldn't cause issues for `rb_profile_frames()` (stackprof) because it uses `VM_FRAME_RUBYFRAME_P`, which checks the frame type instead of just the pc field.

cc @ko1 since this touches the VM a tiny bit. The general direction YJIT wants to move in is to make things working with control frames to check frame flag and using it to know how to read certain data (sort of like a tagged union). I think it's reasonable because it's already done this way in a lot of cases with `rb_profile_frames()` being an example.

File: yjit/src/codegen.rs [hunk 1]
```diff
@@ -5095,7 +5095,11 @@ fn gen_send_cfunc(
         cme,
         recv,
         sp,
-        pc: Some(0),
+        pc: if cfg!(debug_assertions) {
+            Some(!0) // Poison value. Helps to fast fast.
```
Comment (k0kubun) [inline yjit/src/codegen.rs:5099]: What does "fast fast" mean? Did you mean something like "detect fast"?

**Review by k0kubun — APPROVED** [review]
I'm also looking forward to obviating `__bp__` :wink:

File: yjit/src/codegen.rs (hunk 1 above)
Comment (XrXr) [reply yjit/src/codegen.rs]: Sorry, I meant "fail fast":  b235552

**Pull request merged**